gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageWriteChannel.java

```diff
@@ -83,8 +83,8 @@ public long size() throws IOException {
 
   @Override
   public SeekableByteChannel truncate(long newSize) throws IOException {
-    // TODO(b/18997913): Emulate this functionality by closing and rewriting old file up to newSize.
-    //                   Or maybe just swap out GcsService for the Apiary client.
+    // TODO: Emulate this functionality by closing and rewriting old file up to newSize.
+    //       Or maybe just swap out GcsStorage for the API client.
     throw new UnsupportedOperationException();
   }
 
```

Comment:
I think we should change our code to throw `ClosedChannelException` instead of `IOException("stream is closed")`
and then you will not need the `checkOpen` method.  Do you mind to file a bug for it.
/cc @mziccard